### PR TITLE
Do not consider the spurious blank lines of a file

### LINKED ISSUE #143: Sloc shouldn't count spurious blank lines after the code
It would be possible to create a file which has a series of blank lines after the last line of code.

### PR DESCRIPTION
**Pull request by Luni-4**
This PR fixes #143 

Thanks in advance for your review! :)

### CODE REVIEW

**Comment by Luni-4** [comment]
@marco-c 

All tests here are coherent with the fix, they are necessary. So we can create a single commit for this one

**Comment by marco-c** [comment]
> All tests here are coherent with the fix, they are necessary. So we can create a single commit for this one

Well, they are not all necessary for the fix. E.g. merging the tests into a single test is not necessary.
Please make self-contained PRs, so they are easier to review and easier to backout individually should they cause problems.

**Comment by Luni-4** [comment]
> > All tests here are coherent with the fix, they are necessary. So we can create a single commit for this one
> 
> Well, they are not all necessary for the fix. E.g. merging the tests into a single test is not necessary.
> Please make self-contained PRs, so they are easier to review and easier to backout individually should they cause problems.

Done

File: src/tools.rs [hunk 1]
```diff
@@ -243,6 +239,13 @@ pub fn guess_language<P: AsRef<Path>>(buf: &[u8], path: P) -> (Option<LANG>, Str
     }
 }
 
+pub(crate) fn remove_blank_lines(data: &mut Vec<u8>) {
+    while data.last().map_or_else(|| false, |c| *c == b'\n') {
```
Comment (calixteman) [inline src/tools.rs:243]: You could just count the trailing \n and then `resize` the vector correctly or push a new \n if no \n.

File: src/tools.rs [hunk 2]
```diff
@@ -243,6 +239,15 @@ pub fn guess_language<P: AsRef<Path>>(buf: &[u8], path: P) -> (Option<LANG>, Str
     }
 }
 
+pub(crate) fn remove_blank_lines(data: &mut Vec<u8>) {
```
Comment (calixteman) [inline src/tools.rs:242]: I'd say that 99.9% of the time the file has at least one trailing EOL. So I think we should compute `count_trailing` and then if !0 then `truncate` (which is better than resize) else push an EOL.

**Review by calixteman — APPROVED** [review]

**Pull request merged**